src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2933,40 +2913,40 @@ export class StudyViewPageStore {
 
     public async getSurvivalDownloadData(chartMeta: ChartMeta)
     {
-        if (this.survivalData.result)
+        const matchedPlot = _.find(this.survivalPlots, plot =>plot.id=chartMeta.uniqueKey);
+        if (matchedPlot && this.survivalData.result)
         {
             const data: string[] = [];
 
             // find the unique clinical attribute ids
-            const uniqueClinicalAttributeIds: {[clinicalAttributeId: string]: string} = _.reduce(
-                this.survivalData.result,
-                (map: {[clinicalAttributeId: string]: string}, clinicalDataList: ClinicalData[]) => {
-                    clinicalDataList.forEach(clinicalData => {
-                        if (clinicalData.clinicalAttributeId) {
-                            map[clinicalData.clinicalAttributeId] = clinicalData.clinicalAttributeId;
-                        }
-                    });
-
-                    return map;
-                },
-                {}
-            );
+            const uniqueClinicalAttributeIds = matchedPlot.associatedAttrs;
 
             // add the header row
-            data.push(["Study ID", "Patient ID", ..._.values(uniqueClinicalAttributeIds)].join("\t"));
+            data.push(["Study ID", "Patient ID", ...uniqueClinicalAttributeIds].join("\t"));
 
             // add the data rows
-            _.each(this.survivalData.result, clinicalDataList => {
+            this.selectedPatientKeys.result.forEach(uniquePatientKey => {
+                const clinicalDataList = this.survivalData.result[uniquePatientKey];
                 const row: string[] = [];
 
-                if (clinicalDataList.length > 0) {
+                if (clinicalDataList && clinicalDataList.length > 0) {
                     row.push(clinicalDataList[0].studyId || NA_DATA);
                     row.push(clinicalDataList[0].patientId || NA_DATA);
                     const keyed = _.keyBy(clinicalDataList, 'clinicalAttributeId');
 
                     _.each(uniqueClinicalAttributeIds, id => {
                         row.push(keyed[id] ? keyed[id].value || NA_DATA : NA_DATA);
                     });
+                } else {
+                    const selectedPatient = _.find(this.selectedPatients, selectedPatient => selectedPatient.uniquePatientKey === uniquePatientKey);
```

Comment:
you're doing linear search within a linear loop, meaning quadratic time. should use a map from unique patient key to patient here for the lookup, instead of using _.find. would otherwise be very slow for lots of patients.